### PR TITLE
Add joystick support for antipodes builds

### PR DESCRIPTION
**Pull request by asarium**
This adds SDL2 as a dependency which replaces SDL 1.2. I haven't tested the changes on linux yet but it's likely that compilation is broken as SDL2 is not backwards compatible.
I'd like to know if my approach for supporting multiple joystick APIs is acceptable, if it is then I'll make the necessary changes for linux.

### CODE REVIEW

File: code/apis/JoystickManager.cpp [hunk 1]
```diff
@@ -75,9 +96,12 @@ bool JoyMan::Initialize() {
...
 #endif
 
-#if USE_JOYSTICK && IS_WIN32
+#if USE_JOYSTICK
+#if IS_WIN32
+	if (currentApi == API_NATIVE)
+	{
 	UINT num = joyGetNumDevs(); // get the number of joys supported by windows.
-	if ( num > 16 ) {
+		if (num > 16) {
```
Comment (IssMneur) [inline code/apis/JoystickManager.cpp]: This change in whitespace doesn't make sense.

File: code/apis/JoystickManager.cpp [hunk 2]
```diff
@@ -89,57 +113,77 @@ bool JoyMan::Initialize() {
...
+
 	for (UINT counter = 0; counter < num; counter++) {
 		memset(reinterpret_cast<void*>(&joyinfo), 0, sizeof(JOYINFO));
-		
+
 		result = joyGetPos(counter, &joyinfo);
 		wxString joystickName;
 
-		if ( result == JOYERR_NOERROR ) {
+			if (result == JOYERR_NOERROR) {
```
Comment (IssMneur) [inline code/apis/JoystickManager.cpp]: Same with this whitespace change.

**Comment by IssMneur** [comment]
The approach looks good to me.

Three issues I see:
* Please move the libsdl addition to a separate commit with an appropriate commit message.
* Please move the libsdl headers to a `3rdparty` folder and put the version number of the headers with the periods replaced with `_`
* If possible do the whitespace changes in a separate commit as well.

**Comment by asarium** [comment]
I have no idea what was up with that whitespace, should be fixed now. The SDL headers now have a version number and are located in a subfolder.

If the general concept is good then I'll try to get it to work on linux but I don't have a Mac so I can't test that. Hopefully linux compatibility also covers Mac...

**Comment by asarium** [comment]
I changed the display mode detection to use the SDL2 API but for some reason the launcher now crashes on startup which doesn't happen with the 0.9.6 branch, I'll have to do some more testing.

**Pull request merged**